pkg/e2e/infra/infra.go

```diff
@@ -197,243 +223,70 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
-		}()
-
-		g.By("Taking the first worker machineset (assuming only worker machines are backed by machinesets)")
-		machinesets := mapiv1beta1.MachineSetList{}
-		err = wait.PollImmediate(e2e.RetryMedium, e2e.WaitShort, func() (bool, error) {
-			if err := client.List(context.TODO(), &machinesets); err != nil {
-				glog.Errorf("Error querying api for machineset object: %v, retrying...", err)
-				return false, nil
-			}
-			if len(machinesets.Items) < 1 {
-				glog.Errorf("Expected at least one machineset, have none")
-				return false, nil
-			}
-			return true, nil
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By("Create a machine for node about to be drained")
 
-		g.By("Creating two new machines, one for node about to be drained, other for moving workload from drained node")
-		clusterInfra, err := e2e.GetInfrastructure(client)
-		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(clusterInfra.Status.InfrastructureName).ShouldNot(o.BeEmpty())
-
-		// Create two machines
-		randID := e2e.RandomString()
-		machine1 := machineFromMachineset(&machinesets.Items[0])
-		machine1.Name = fmt.Sprintf("%s-machine1-%s",
-			clusterInfra.Status.InfrastructureName,
-			randID,
-		)
-
-		var machine2 *mapiv1beta1.Machine
-		err = func() error {
-			if err := client.Create(context.TODO(), machine1); err != nil {
-				return fmt.Errorf("unable to create machine %q: %v", machine1.Name, err)
-			}
-			delObjects["machine1"] = machine1
+		selector := machineSet.Spec.Selector
+		machines, err := e2e.GetMachines(client, &selector)
+		o.Expect(err).ToNot(o.HaveOccurred())
 
-			machine2 = machineFromMachineset(&machinesets.Items[0])
-			machine2.Name = "machine2" + randID
-			machine2.Name = fmt.Sprintf("%s-machine2-%s",
-				clusterInfra.Status.InfrastructureName,
-				randID,
-			)
+		// Add node draining labels to params
+		for k, v := range nodeDrainLabels {
+			machineSetParams.Labels[k] = v
+		}
 
-			if err := client.Create(context.TODO(), machine2); err != nil {
-				return fmt.Errorf("unable to create machine %q: %v", machine2.Name, err)
-			}
-			delObjects["machine2"] = machine2
+		machines[0].Spec.ObjectMeta.Labels = machineSetParams.Labels
+		machines[1].Spec.ObjectMeta.Labels = machineSetParams.Labels
 
-			return nil
-		}()
-		o.Expect(err).NotTo(o.HaveOccurred())
+		err = client.Update(context.TODO(), machines[0])
+		o.Expect(err).ToNot(o.HaveOccurred())
 
-		g.By("Waiting until both new nodes are ready")
-		err = waitUntilNodesAreReady(client, []runtimeclient.ListOption{runtimeclient.MatchingLabels(nodeDrainLabels)}, 2)
-		o.Expect(err).NotTo(o.HaveOccurred())
+		err = client.Update(context.TODO(), machines[1])
+		o.Expect(err).ToNot(o.HaveOccurred())
 
 		g.By("Creating RC with workload")
 		rc := replicationControllerWorkload("default")
 		err = client.Create(context.TODO(), rc)
 		o.Expect(err).NotTo(o.HaveOccurred())
-		delObjects["rc"] = rc
 
 		g.By("Creating PDB for RC")
 		pdb := podDisruptionBudget("default")
```

Comment:
can we please make sure this is always deleted before the end of the test as it was via delObjects